packages/client/screens/AuthenticationSignUpForm/index.tsx

```diff
@@ -0,0 +1,489 @@
+import { SafeAreaView, Text, useSx, View } from 'dripsy';
+import React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { TouchableOpacity } from 'react-native';
+import * as z from 'zod';
+import { createMachine } from 'xstate';
+import { useMachine } from '@xstate/react';
+import Toast from 'react-native-toast-message';
+import { passwordStrengthRegex } from '@musicroom/types';
+import { assertEventType } from '../../machines/utils';
+import { AppScreen, TextField } from '../../components/kit';
+import { sendSignUp, SignUpError } from '../../services/AuthenticationService';
+import { SignUpFormScreenProps } from '../../types';
+
+export interface AuthenticationSignUpFormFormFieldValues {
+    userNickname: string;
+    password: string;
+    email: string;
+}
+
+type UpdateNicknameMachineEvent =
+    | {
+          type: 'Validated sign up form';
+          body: AuthenticationSignUpFormFormFieldValues;
+      }
+    | { type: 'Signed up successfully' }
+    | { type: 'Nickname unavailable' }
+    | { type: 'Email unavailable' }
+    | { type: 'Email invalid' }
+    | { type: 'Weak password' }
+    | { type: 'Unknown error' };
+
+const signUpMachine =
+    /** @xstate-layout N4IgpgJg5mDOIC5QFUAOECGAXMACAdgJYDGA1vhgLZgB0AYmFsQBaH5S4CusYATgOSxcPLFjZRYAYggB7fLTYA3GaVppMOAiXJVaDJq3ZceAoSLHtYCJTOLZCcgNoAGALqJQqGbEJi5HkAAPRAA2ZwAmGgAWAA5nAE4Y+IBGKNSAVmSYgBoQAE9QmIBmGhCYgHYQ9PS0mPSw5IBfRtz1bDwiMgpqekYWcWM+QWFGCwlJPl4ZXhpUABtsADNpyho2zU6dHv1+o24hs1HxKxs7P3wXdyQQLx9zgOCESpiaROciorjwmPCQkPLcgUEEkSvUYnFPuVylEos1Wuh2louroaABJCBzMCSABqGDmhA0kFwy14lACt18DnwD0QNWcNGS8T+UWhyXS8Q5RUBiHi9I5zgq5VS4SKUQF6ThIHWHW03VoAGUwPgIANNnLcFgZCNeIo+JJpRAkVs8LBOMRiHBYItOHM5kDPN5Kf5ro9PlEaOUmezwvEiiEotVuQg2e6heFMvFyuzkkURZLpUa5TRFcrVbLdBqtSZdbx9SZhsROLxeEqsIndOTHfcXYg3R6vfEfX6A+kgz70jQKoz-clvrEffGERt0z0UyqjGqM5rtTn9fhyDIAO74XCTaaVu5UmkIOueqqN33+wP5RBijtFZK95xhKKJdLhkKDwnl0dK8ccSfUTMzvUAORHeCcBQigYIQCwAEaYhuTrUjWO4xKGDZNkerYnggmShukzgwp6YbhHGLRSkOMrIj0AAiYDgTIQHEAM+ymLgbAkpQ9hyMSfSGFAkiBLAWDtDQGCLDgMxFM4zjQdWoCPOUMQhNE4QBuGNSXopQYwiUzjQvy-wfGUBGEfgMgQHAAQJp+egcXR+aHKIxwSVucGih2ISJN8vY+vhsRBryJSyf6nr3lEnypE+iLmTQACixbTIxK70cM5jHOxBjiPZzpSTyUSRGG-ZJLyUbhEGUKRM47lih8WXpFCoXDqRtDolB1wUpJQSIOUPoeuUolVOUmm8oyQaxuUNCiZG-zpKKmn4TVJHGsmb5pnV37ZnwaWwRlCBZd58Q0NeZRJGUUbpAhsKEWZAE0BRVE0VZByxcxrEroslnsGt26VO6bpFFV16Mj8hVoTUkT4Qhnrhsk-oJDNL5gG9cFZEGyTDRyHJJFEZQZBDzTNEAA */
+    createMachine<Record<string, never>, UpdateNicknameMachineEvent>({
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Did you create your machine by using Stately Visual Editor?

Reply:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏I really like your machine!

Reply:
Thanks ! not really, the next one will be